azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -260,7 +260,7 @@ func resourceArmKubernetesClusterRead(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Error setting `linux_profile`: %+v", err)
 	}
 
-	agentPoolProfiles := flattenAzureRmKubernetesClusterAgentPoolProfiles(resp.ManagedClusterProperties.AgentPoolProfiles)
+	agentPoolProfiles := flattenAzureRmKubernetesClusterAgentPoolProfiles(resp.ManagedClusterProperties.AgentPoolProfiles, *resp.Fqdn)
```

Comment:
In the unlikely situation where `resp.Fqdn` is null this will crash. I suggest passing in the pointer and then checking it for null like all the other values in `agent_pool_profile` and dereferencing it there.

Reply:
Good catch. Updated